hi-ml/src/health_ml/utils/checkpoint_utils.py

```diff
@@ -178,6 +201,9 @@ def find_checkpoint_with_highest_epoch(files: List[Path]) -> Optional[Path]:
                 epoch = _load_epoch_from_checkpoint(file)
                 logging.info(f"Checkpoint for epoch {epoch} in {file}")
                 if (highest_epoch is None) or (epoch > highest_epoch):
+                    if file_with_highest_epoch is not None and delete_files:
```

Comment:
High level of indentation is hurting readability here. Simple refactor of putting everything in the `try` statement into a separate function would help I think.